### ISSUE TITLE
Compile issue on OSX / M1

### ISSUE THREAD

**Issue by rob-p** [rust-bio/rust-htslib#352]
We're seeing a compilation issue in rust-htslib as a dependency when building on OSX; any insight on [this](https://github.com/OceanGenomics/mudskipper/issues/23)?  Mostly many similar errors to this one:

```
#0 131.0 error[E0308]: mismatched types
#0 131.0    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-htslib-0.38.2/src/bam/record.rs:152:16
#0 131.0     |
#0 131.0 152 |             s: sam_copy.as_ptr() as *mut i8,
#0 131.0     |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
#0 131.0     |
#0 131.0     = note: expected raw pointer `*mut u8`
#0 131.0                found raw pointer `*mut i8`
#0 131.0 
```

**Comment by jch-13** [comment]
I think this has been fixed in `master` already: a21aff289bc03c7549afc7a958084ed57e8c93f2 . Should be in the next release!

**Comment by rob-p** [comment]
Awesome; thanks!  This is now in the new tagged release.
